oci/layout/oci_archive_dest.go

```diff
@@ -0,0 +1,126 @@
+package layout
```

Comment:
I started out by packaging this separately, but I was having a hard time initializing ociReference. Since all the functions of oci-archive call the functions in oci, I need to be able to set ociReference and use it to call the oci functions. I was using Get and Set functions but was failing to get it work. 
Things make more sense now, so I am going to work on separating it.